From 25a07c1c58dab6e2409b060bd00b65cab909b189 Mon Sep 17 00:00:00 2001
From: Martchus <martchus@gmx.net>
Date: Sun, 25 Sep 2016 21:44:42 +0200
Subject: [PATCH 1/2] Fix linguist macro

- Prevent CMake appending extra 'Qt5::lupdate'/'Qt5::lrelease' to
  command line when invoking lupdate/lrelease
- Achieved by resolveing the IMPORTED_LOCATION manually
---
 src/linguist/Qt5LinguistToolsMacros.cmake | 7 +++++--
 1 file changed, 5 insertions(+), 2 deletions(-)

diff --git a/src/linguist/Qt5LinguistToolsMacros.cmake b/src/linguist/Qt5LinguistToolsMacros.cmake
index 8e363a40..d7fd0f8b 100644
--- a/src/linguist/Qt5LinguistToolsMacros.cmake
+++ b/src/linguist/Qt5LinguistToolsMacros.cmake
@@ -71,8 +71,10 @@ function(QT5_CREATE_TRANSLATION _qm_files)
 
           file(WRITE ${_ts_lst_file} "${_lst_file_srcs}")
         endif()
+
+        get_target_property(LUPDATE_LOC ${Qt5_LUPDATE_EXECUTABLE} IMPORTED_LOCATION)
         add_custom_command(OUTPUT ${_ts_file}
-            COMMAND ${Qt5_LUPDATE_EXECUTABLE}
+            COMMAND ${LUPDATE_LOC}
             ARGS ${_lupdate_options} "@${_ts_lst_file}" -ts ${_ts_file}
             DEPENDS ${_my_sources} ${_ts_lst_file} VERBATIM)
     endforeach()
@@ -95,8 +97,9 @@ function(QT5_ADD_TRANSLATION _qm_files)
             set(qm "${CMAKE_CURRENT_BINARY_DIR}/${FILE_NAME}.qm")
         endif()
 
+        get_target_property(LRELEASE_LOC ${Qt5_LRELEASE_EXECUTABLE} IMPORTED_LOCATION)
         add_custom_command(OUTPUT ${qm}
-            COMMAND ${Qt5_LRELEASE_EXECUTABLE}
+            COMMAND ${LRELEASE_LOC}
             ARGS ${_abs_FILE} -qm ${qm}
             DEPENDS ${_abs_FILE} VERBATIM
         )
-- 
2.16.1

